lib/livebook/notebook/explore/elixir_and_livebook.livemd

```diff
@@ -1,60 +1,84 @@
 # Elixir and Livebook
 
-## Modules
+## Introduction
 
-You can use code cells to execute any Elixir code:
+In this notebook, we will explore some unique features when
+using Elixir and Livebook together, such as inputs, autocompletion,
+and more. To get started, execute the cell below so we can get
+the Elixir runtime up and running:
 
 ```elixir
-IO.puts("hello world!")
+"Hello world"
 ```
 
-But you can define modules inside cells too!
+If you are not familiar with Elixir, there is a fast paced
+introduction to the language in the [Distributed portals with
+Elixir](/explore/notebooks/distributed-portals-with-elixir)
+notebook.
 
-```elixir
-defmodule Utils do
-  @doc """
-  Generates a random binary id.
-  """
-  @spec random_id() :: binary()
-  def random_id() do
-    :crypto.strong_rand_bytes(20) |> Base.encode32(case: :lower)
-  end
-end
-```
+Let's move on.
 
-If you're surprised by the above output, keep in mind that
-every Elixir expression evaluates to some value and as so
-does module compilation!
+## Inputs
 
-Having the module defined, let's take it for a spin.
+Livebook supports inputs and you read the input values directly
+from your notebook code, using `IO.gets/1`. Let's see an example
+that expects a date in the format `YYYY-MM-DD` and returns if the
+data is valid or not:
+
+<!-- livebook:{"livebook_object":"cell_input","name":"Date","type":"text","value":""} -->
 
 ```elixir
-Utils.random_id()
+# Read the date input, which returns something like "2020-02-30\n"
+input = IO.gets("Date: ")
+
+# So we trim the newline from the input value
+trimmed = String.trim(input)
+
+# And then match on the return value
+case Date.from_iso8601(trimmed) do
+  {:ok, date} ->
+    "We got a valid data: #{inspect(date)}"
+
+  {:error, reason} ->
+    "Oh no, the date is invalid. Reason: #{inspect(reason)}"
+end
 ```
 
+The string passed to `IO.gets/1` must have the same suffix as the
+input name and the returned string is always appended by a newline.
+This is built-in on top of Erlang's IO protocol and built in a way
+that your notebooks can be exported to Elixir scripts and still work!
+
+Create your own inputs to learn more about the available input types
+and options.
+
 ## Autocompletion
 
 Elixir code cells also support autocompletion by pressing
-<kbd>Ctrl</kbd> + <kbd>Spacebar</kbd>. You can try it out by making sure the
-module in the previous section has been defined and then
-put the cursor after the `.` below and press <kbd>Ctrl</kbd> + <kbd>Spacebar</kbd>:
+<kbd>Ctrl</kbd> + <kbd>Spacebar</kbd>. Try it out by autocompleting
+the code below to `System.version()`. First put the cursor after
+the `.` below and press <kbd>Ctrl</kbd> + <kbd>Spacebar</kbd>:
```

Comment:
We don't want to start the line with the tag, as earmark ignores the rest of the line. So this could be:

```
Elixir code cells also support autocompletion by pressing <kbd>Ctrl</kbd> + <kbd>Spacebar</kbd>.
Try it out by autocompleting the code below to `System.version()`.
First put the cursor after the `.` below and press <kbd>Ctrl</kbd> + <kbd>Spacebar</kbd>:
```